packages/ui/build.mjs

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable import/no-nodejs-modules */
+import { execSync } from 'node:child_process';
+import { createHash } from 'node:crypto';
+import fs from 'node:fs';
+import { sassPlugin, postcssModules } from 'esbuild-sass-plugin';
+import { build } from 'tsup';
+
+let gitHeadSha;
+try {
+	gitHeadSha = execSync( 'git rev-parse HEAD' ).toString();
+} catch {}
+
+// Get the exports from the package.json
+const packageJson = JSON.parse( fs.readFileSync( 'package.json', 'utf8' ) );
+const calypsoSrcs = Object.entries( packageJson.exports )
+	.map( ( [ key, value ] ) => [ key, value[ 'calypso:src' ] ] )
+	.filter( ( [ _key, value ] ) => !! value );
+
+calypsoSrcs.forEach( ( [ entryKey, entryValue ] ) => {
+	build( {
+		entry: [ entryValue ],
+		clean: true,
+		splitting: true,
+		dts: {
+			compilerOptions: {
+				composite: false,
+			},
+		},
```

Comment:
`experimentalDts` was emitting dts files in a nested folder and I couldn't resolve it. This was the only config that I could get to work 🙁